src/lasp_core.erl

```diff
@@ -912,6 +912,34 @@ write(Type, Value, Key, Store) ->
 increment_counter(Counter) ->
     Counter + 1.
 
+%% @private
+store_delta(Type, Counter, Delta, DeltaMap0) ->
+    MaxDeltaSlots = application:get_env(lasp, delta_mode_max_slots, 10),
+    %% Check the space of the DeltaMap
+    case orddict:size(DeltaMap0) < MaxDeltaSlots of
+        true ->
+            %% Store a new delta.
+            orddict:store(Counter, Delta, DeltaMap0);
+        false ->
+            DeltaCounterList = orddict:fetch_keys(DeltaMap0),
+            %% Find the minimum counter & its delta.
+            MinCounter0 = lists:nth(1, DeltaCounterList),
+            MinCounterDelta0 = orddict:fetch(MinCounter0, DeltaMap0),
+            %% Find the 2nd minimum counter & its delta.
+            MinCounter1 = lists:nth(2, DeltaCounterList),
+            MinCounterDelta1 = orddict:fetch(MinCounter1, DeltaMap0),
+            %% Merge them.
+            Merged = lasp_type:merge(Type,
+                                     MinCounterDelta0,
+                                     MinCounterDelta1),
+            %% Store the merged delta (minimum + 2nd minimum).
+            DeltaMap1 = orddict:store(MinCounter0, Merged, DeltaMap0),
+            %% Remove the 2nd minimum delta.
+            DeltaMap2 = orddict:erase(MinCounter1, DeltaMap1),
+            %% Store a new delta.
+            orddict:store(Counter, Delta, DeltaMap2)
```

Comment:
So, it appears the general idea here is that every single time an update is performed, the earliest delta is collapsed in to the next one to open a slot for the newest update, is that correct?